swap/swap_test.go

```diff
@@ -1818,6 +1819,58 @@ func TestPeerGetLastSentCumulativePayout(t *testing.T) {
 	}
 }
 
+func TestAvailableBalance(t *testing.T) {
+	testBackend := newTestBackend()
+	defer testBackend.Close()
+	swap, clean := newTestSwap(t, ownerKey, testBackend)
+
+	// deposit 100
+	depositAmount := big.NewInt(100)
+	opts := bind.NewKeyedTransactor(swap.owner.privateKey)
+	opts.Context = context.TODO()
+	_, err := swap.contract.Deposit(opts, swap.backend, depositAmount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var stb *swapTestBackend
+	var ok bool
+	if stb, ok = swap.backend.(*swapTestBackend); !ok {
+		t.Fatal(errors.New("not the expected test backend"))
+	}
+	stb.Commit()
+
+	// verify that available balance equals depositAmount
+	availableBalance, err := swap.AvailableBalance()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if availableBalance != depositAmount.Uint64() {
+		t.Fatalf("availableBalance not equal to deposited amount. availableBalance: %d, depositAmount: %d", availableBalance, depositAmount.Uint64())
+	}
+
+	// withdraw 50
+	withdrawAmount := big.NewInt(50)
+	_, err = swap.contract.Withdraw(opts, swap.backend, withdrawAmount)
```

Comment:
same thing about the receipt here

Reply:
Added a test that verifies that the transaction did not revert.